Remotion/Data/DomainObjects/Persistence/StorageProviderManager.cs

```diff
@@ -8,24 +8,25 @@
 // 
 // re-motion is distributed in the hope that it will be useful, 
 // but WITHOUT ANY WARRANTY; without even the implied warranty of 
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 // GNU Lesser General Public License for more details.
 // 
 // You should have received a copy of the GNU Lesser General Public License
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using Remotion.Data.DomainObjects.Persistence.Configuration;
 using Remotion.Data.DomainObjects.Tracing;
 using Remotion.Utilities;
 
 namespace Remotion.Data.DomainObjects.Persistence
 {
-public class StorageProviderManager : IDisposable
+public class StorageProviderManager : IStorageProviderManager, IReadOnlyStorageProviderManager
 {
   private bool _disposed;
-  private StorageProviderCollection? _storageProviders;
+  private Dictionary<string, IStorageProvider>? _storageProviders;
```

Comment:
This field should be non-nullable. In `Dispose()`, we can just clear the dictionary - no need to set it to `null`.